File: cSharp/Api-ng-sample-code/Api-ng-sample-code/TO/MarketDescription.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Newtonsoft.Json;

namespace Api_ng_sample_code.TO
{
    public class MarketDescription
    {
        [JsonProperty(PropertyName = "persistenceEnabled")]
        public bool IsPersistenceEnabled { get; set; }

        [JsonProperty(PropertyName = "bspMarket")]
        public bool IsBspMarket { get; set; }

        [JsonProperty(PropertyName = "marketTime")]
        public DateTime MarketTime { get; set; }

        [JsonProperty(PropertyName = "suspendTime")]
        public DateTime? SuspendTime { get; set; }

        [JsonProperty(PropertyName = "settleTime")]
        public DateTime? SettleTime { get; set; }

        [JsonProperty(PropertyName = "bettingType")]
        public MarketBettingType BettingType { get; set; }

        [JsonProperty(PropertyName = "turnInPlayEnabled")]
        public bool IsTurnInPlayEnabled { get; set; }

        [JsonProperty(PropertyName = "marketType")]
        public string MarketType { get; set; }

        [JsonProperty(PropertyName = "regulator")]
        public string Regulator { get; set; }

        [JsonProperty(PropertyName = "marketBaseRate")]
        public double MarketBaseRate { get; set; }

        [JsonProperty(PropertyName = "discountAllowed")]
        public bool IsDiscountAllowed { get; set; }

        [JsonProperty(PropertyName = "wallet")]
        public string Wallet { get; set; }

        [JsonProperty(PropertyName = "rules")]
        public string Rules { get; set; }

        [JsonProperty(PropertyName = "rulesHasDate")]
        public bool RulesHasDate { get; set; }

        [JsonProperty(PropertyName = "clarifications")]
        public string Clarifications { get; set; }

        public override string ToString()
        {
            return new StringBuilder().AppendFormat("{0}", "MarketDescription")
                        .AppendFormat(" : BspMarket={0}", IsBspMarket)
                        .AppendFormat(" : BettingType={0}", BettingType)
                        .AppendFormat(" : MarketType={0}", MarketType)

                        .AppendFormat(" : MarketTime={0}", MarketTime)
                        .AppendFormat(" : SuspendTime={0}", SuspendTime)
                        .AppendFormat(" : SettleTime={0}", SettleTime)
                        .AppendFormat(" : MarketBaseRate={0}", MarketBaseRate)

                        .AppendFormat(" : IsPersistenceEnabled={0}", IsPersistenceEnabled)
                        .AppendFormat(" : IsTurnInPlayEnabled={0}", IsTurnInPlayEnabled)
                        .AppendFormat(" : IsDiscountAllowed={0}", IsDiscountAllowed)

                        .AppendFormat(" : Regulator={0}", Regulator)
                        .AppendFormat(" : Rules={0}", Rules)
                        .AppendFormat(" : RulesHasDate={0}", RulesHasDate)
                        .AppendFormat(" : Clarifications={0}", Clarifications)

                        .AppendFormat(" : Wallet={0}", Wallet)
                        .ToString();
        }
    }
}

File: android/src/com/betfair/api/ng/objects/MarketDescription.java
package com.betfair.api.ng.objects;

import java.util.Date;

import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

public class MarketDescription {
	
	 private Boolean persistenceEnabled;
	    private Boolean bspMarket;
	    private Date marketTime;
	    private Date suspendTime;
	    private Date settleTime;
	    private String bettingType;
	    private Boolean turnInPlayEnabled;
	    private String marketType;
	    private String regulator;
	    private Double marketBaseRate;
	    private Boolean discountAllowed;
	    private String wallet;
	    private String rules;
	    private Boolean rulesHasDate;
	    private String clarifications;
	    
	    
	    public Boolean getPersistenceEnabled() {
			return persistenceEnabled;
		}

		public void setPersistenceEnabled(Boolean persistenceEnabled) {
			this.persistenceEnabled = persistenceEnabled;
		}

		public Boolean getBspMarket() {
			return bspMarket;
		}

		public void setBspMarket(Boolean bspMarket) {
			this.bspMarket = bspMarket;
		}

		public Date getMarketTime() {
			return marketTime;
		}

		public void setMarketTime(Date marketTime) {
			this.marketTime = marketTime;
		}

		public Date getSuspendTime() {
			return suspendTime;
		}

		public void setSuspendTime(Date suspendTime) {
			this.suspendTime = suspendTime;
		}

		public Date getSettleTime() {
			return settleTime;
		}

		public void setSettleTime(Date settleTime) {
			this.settleTime = settleTime;
		}

		public String getBettingType() {
			return bettingType;
		}

		public void setBettingType(String bettingType) {
			this.bettingType = bettingType;
		}

		public Boolean getTurnInPlayEnabled() {
			return turnInPlayEnabled;
		}

		public void setTurnInPlayEnabled(Boolean turnInPlayEnabled) {
			this.turnInPlayEnabled = turnInPlayEnabled;
		}

		public String getMarketType() {
			return marketType;
		}

		public void setMarketType(String marketType) {
			this.marketType = marketType;
		}

		public String getRegulator() {
			return regulator;
		}

		public void setRegulator(String regulator) {
			this.regulator = regulator;
		}

		public Double getMarketBaseRate() {
			return marketBaseRate;
		}

		public void setMarketBaseRate(Double marketBaseRate) {
			this.marketBaseRate = marketBaseRate;
		}

		public Boolean getDiscountAllowed() {
			return discountAllowed;
		}

		public void setDiscountAllowed(Boolean discountAllowed) {
			this.discountAllowed = discountAllowed;
		}

		public String getWallet() {
			return wallet;
		}

		public void setWallet(String wallet) {
			this.wallet = wallet;
		}

		public String getRules() {
			return rules;
		}

		public void setRules(String rules) {
			this.rules = rules;
		}

		public Boolean getRulesHasDate() {
			return rulesHasDate;
		}

		public void setRulesHasDate(Boolean rulesHasDate) {
			this.rulesHasDate = rulesHasDate;
		}

		public String getClarifications() {
			return clarifications;
		}

		public void setClarifications(String clarifications) {
			this.clarifications = clarifications;
		}

		public String toString() {
	    	return "{"+""+"persistenceEnabled="+getPersistenceEnabled()+","+"bspMarket="+getBspMarket()+","+"marketTime="+getMarketTime()+","+"suspendTime="+getSuspendTime()+","+"settleTime="+getSettleTime()+","+"bettingType="+getBettingType()+","+"turnInPlayEnabled="+getTurnInPlayEnabled()+","+"marketType="+getMarketType()+","+"regulator="+getRegulator()+","+"marketBaseRate="+getMarketBaseRate()+","+"discountAllowed="+getDiscountAllowed()+","+"wallet="+getWallet()+","+"rules="+getRules()+","+"rulesHasDate="+getRulesHasDate()+","+"clarifications="+getClarifications()+","+"}";
	    }

	    public boolean equals(Object o) {
	        if (!(o instanceof MarketDescription)) {
	            return false;
	        }

	        if (this == o) {
	            return true;
	        }
	        MarketDescription another = (MarketDescription)o;

	        return new EqualsBuilder()
	            .append(persistenceEnabled, another.persistenceEnabled)
	            .append(bspMarket, another.bspMarket)
	            .append(marketTime, another.marketTime)
	            .append(suspendTime, another.suspendTime)
	            .append(settleTime, another.settleTime)
	            .append(bettingType, another.bettingType)
	            .append(turnInPlayEnabled, another.turnInPlayEnabled)
	            .append(marketType, another.marketType)
	            .append(regulator, another.regulator)
	            .append(marketBaseRate, another.marketBaseRate)
	            .append(discountAllowed, another.discountAllowed)
	            .append(wallet, another.wallet)
	            .append(rules, another.rules)
	            .append(rulesHasDate, another.rulesHasDate)
	            .append(clarifications, another.clarifications)
	            .isEquals();
	    }

	    public int hashCode() {
	        return new HashCodeBuilder()
	            .append(persistenceEnabled)
	            .append(bspMarket)
	            .append(marketTime)
	            .append(suspendTime)
	            .append(settleTime)
	            .append(bettingType)
	            .append(turnInPlayEnabled)
	            .append(marketType)
	            .append(regulator)
	            .append(marketBaseRate)
	            .append(discountAllowed)
	            .append(wallet)
	            .append(rules)
	            .append(rulesHasDate)
	            .append(clarifications)
	            .toHashCode();
	    }

}
